include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1321,21 +1321,29 @@ namespace internal
 
         // ---------------------- lexicographic_numbering ----------------------
         std::vector<unsigned int> lexicographic_numbering;
-        if (reference_cell == ReferenceCells::get_hypercube<dim>())
+        if (fe_fine.n_base_elements() == 1 &&
+            fe_fine.base_element(0).n_components() == 1)
           {
-            const Quadrature<1> dummy_quadrature(
-              std::vector<Point<1>>(1, Point<1>()));
-            internal::MatrixFreeFunctions::ShapeInfo<Number> shape_info;
-            shape_info.reinit(dummy_quadrature, fe_fine, 0);
-            lexicographic_numbering = shape_info.lexicographic_numbering;
+            if (reference_cell == ReferenceCells::get_hypercube<dim>())
+              {
+                const Quadrature<1> dummy_quadrature(
+                  std::vector<Point<1>>(1, Point<1>()));
+                internal::MatrixFreeFunctions::ShapeInfo<Number> shape_info;
+                shape_info.reinit(dummy_quadrature, fe_fine, 0);
+                lexicographic_numbering = shape_info.lexicographic_numbering;
+              }
+            else
+              {
+                const auto dummy_quadrature =
+                  reference_cell.template get_gauss_type_quadrature<dim>(1);
+                internal::MatrixFreeFunctions::ShapeInfo<Number> shape_info;
+                shape_info.reinit(dummy_quadrature, fe_fine, 0);
+                lexicographic_numbering = shape_info.lexicographic_numbering;
+              }
           }
         else
           {
-            const auto dummy_quadrature =
-              reference_cell.template get_gauss_type_quadrature<dim>(1);
-            internal::MatrixFreeFunctions::ShapeInfo<Number> shape_info;
-            shape_info.reinit(dummy_quadrature, fe_fine, 0);
-            lexicographic_numbering = shape_info.lexicographic_numbering;
+            AssertThrow(false, ExcNotImplemented());
```

Comment:
Is there an assumption about a single component somewhere else or are you now restricting to a smaller class of finite elements in the hypercube case? The original code of `ShapeInfo` allowed multiple components, but you ask for `n_components()==1` now, which is more restrictive than `fe.n_base_elements()==1` I found as assertions elsewhere.